test/Microsoft.AspNetCore.Identity.Test/SignInManagerTest.cs

```diff
@@ -543,23 +562,39 @@ public override Task<bool> ValidateAsync(string purpose, string token, UserManag
         }
 
         [Theory]
-        [InlineData(true, true, true, true)]
-        [InlineData(true, true, false, true)]
-        [InlineData(true, false, true, true)]
-        [InlineData(true, false, false, true)]
-        [InlineData(false, true, true, true)]
-        [InlineData(false, true, false, true)]
-        [InlineData(false, false, true, true)]
-        [InlineData(false, false, false, true)]
-        [InlineData(true, true, true, false)]
-        [InlineData(true, true, false, false)]
-        [InlineData(true, false, true, false)]
-        [InlineData(true, false, false, false)]
-        [InlineData(false, true, true, false)]
-        [InlineData(false, true, false, false)]
-        [InlineData(false, false, true, false)]
-        [InlineData(false, false, false, false)]
-        public async Task CanTwoFactorSignIn(bool isPersistent, bool supportsLockout, bool externalLogin, bool rememberClient)
+        [InlineData(true, true, true, true, true)]
```

Comment:
I'm not sure how I feel about this block of 32 permutations of true/false, but I guess its a good way to test every permutation :/  But I am not looking forward to any doubling in the future...